packages/editor/src/components/post-actions/reset-posts-page.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useMemo } from '@wordpress/element';
+import {
+	Button,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from '../../utils';
+
+const ResetPostsPageModal = ( { items, closeModal } ) => {
+	const [ item ] = items;
+	const pageTitle = getItemTitle( item );
+	const { isPageOnFrontSet, isSaving } = useSelect( ( select ) => {
+		const { getEntityRecord, isSavingEntityRecord } = select( coreStore );
+		const siteSettings = getEntityRecord( 'root', 'site' );
+		return {
+			isPageOnFrontSet: siteSettings?.page_on_front !== 0,
+			isSaving: isSavingEntityRecord( 'root', 'site' ),
+		};
+	} );
+
+	const { saveEditedEntityRecord, saveEntityRecord } =
+		useDispatch( coreStore );
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+
+	async function onResetPostsPage( event ) {
+		event.preventDefault();
+
+		try {
+			// Save new posts page settings.
+			await saveEditedEntityRecord( 'root', 'site', undefined, {
+				page_for_posts: 0,
+				show_on_front: isPageOnFrontSet ? 'page' : 'posts',
+			} );
+
+			// This second call to a save function is a workaround for a bug in
+			// `saveEditedEntityRecord`. This forces the root site settings to be updated.
+			// See https://github.com/WordPress/gutenberg/issues/67161.
+			await saveEntityRecord( 'root', 'site', {
+				page_for_posts: 0,
+				show_on_front: isPageOnFrontSet ? 'page' : 'posts',
+			} );
```

Comment:
I don't know much about this, but wouldn't just `saveEntityRecord` work correctly?

Reply:
I looked through the codebase some more at other uses of these functions, and `saveEntityRecord` is often used on its own (rather than `saveEditedEntityRecord` as I expected), so I've removed `saveEditedEntityRecord` from both the "Set as homepage" and "Set as posts page" actions. They still work fine so I think this is a nice refactor!

Commit with these changes is here: https://github.com/WordPress/gutenberg/pull/67650/commits/2cfd8172456f9e5679c408519db73efb696f532c.